### PR TITLE
#86: Resolve build warnings

### LINKED ISSUE #86: Resolve build warnings
Building vt-tv produces lots of warnings that are blocking vt [PR 2294](https://github.com/DARMA-tasking/vt/pull/2294)

We should turn on warnings for vttv builds and then resolve them.

### PR DESCRIPTION
**Pull request by cwschilly**
Fixes #86 

This will allow vt's `VT_TV_ENABLED` CI pipeline to pass

### CODE REVIEW

File: cmake/turn_on_warnings.cmake [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  add_cxx_compiler_flag_if_supported("-Wsign-compare")
+  add_cxx_compiler_flag_if_supported("-Wsuggest-override")
+  add_cxx_compiler_flag_if_supported("-pedantic")
+  # Not really a warning, is still diagnostic related..
+  if (NOT CMAKE_CXX_COMPILER_ID STREQUAL Intel OR
+      CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 2021)
+    add_cxx_compiler_flag_if_supported("-ftemplate-backtrace-limit=100")
+  endif()
+
+  if (vt_werror_enabled)   # Treat warning as errors
```
Comment (pierrepebay) [inline cmake/turn_on_warnings.cmake:32]: this is a `vt` specific [build option](=https://darma-tasking.github.io/docs/html/vt-build.html), so we might want to add that as a build option for `vt-tv` as well.

File: cmake/turn_on_warnings.cmake [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    add_cxx_compiler_flag_if_supported("-Werror")
+  endif()
+endif()
+
+set(
+  VT_TV_WARNING_FLAGS ${VT_TV_WARNING_FLAGS} CACHE INTERNAL
+  "Project's warning options")
+
+macro(turn_on_warnings vt_target)
```
Comment (pierrepebay) [inline cmake/turn_on_warnings.cmake:41]: rename `vt_...` to `vt_tv_`

File: examples/CMakeLists.txt [hunk 3]
```diff
@@ -18,6 +18,9 @@ file(
   "${PROJECT_EXAMPLE_DIR}/*.cc"
 )
 
+include(turn_on_warnings)
+turn_on_warnings(${VT_TV_LIBRARY})
```
Comment (pierrepebay) [inline examples/CMakeLists.txt:22]: I see this is already done in `src/CMakeLists.txt`, so do you want to target the examples here instead?

File: tests/CMakeLists.txt [hunk 4]
```diff
@@ -30,6 +30,9 @@ file(
   "${PROJECT_TEST_UNIT_DIR}/*.cc"
 )
 
+include(turn_on_warnings)
+turn_on_warnings(${VT_TV_LIBRARY})
```
Comment (pierrepebay) [inline tests/CMakeLists.txt:34]: same thing here, target tests instead

**Review by pierrepebay — CHANGES_REQUESTED** [review]

File: tests/CMakeLists.txt (hunk 4 above)
Comment (cwschilly) [reply tests/CMakeLists.txt]: Fixed

File: examples/CMakeLists.txt (hunk 3 above)
Comment (cwschilly) [reply examples/CMakeLists.txt]: Fixed

File: cmake/turn_on_warnings.cmake (hunk 2 above)
Comment (cwschilly) [reply cmake/turn_on_warnings.cmake]: Fixed

File: cmake/turn_on_warnings.cmake (hunk 1 above)
Comment (cwschilly) [reply cmake/turn_on_warnings.cmake]: Fixed

**Review by pierrepebay — APPROVED** [review]
looks good, thanks for modifications 👍🏻

**Review by nlslatt — COMMENTED** [review]
This looks good. Should we enable `-Werror` in the CI?

**Comment by cwschilly** [comment]
> This looks good. Should we enable `-Werror` in the CI?

I think we should, especially since vt's `vt-tv` pipeline was failing with `-Werror`. @pierrepebay @tlamonthezie Do you have any thoughts?

**Comment by pierrepebay** [comment]
> > This looks good. Should we enable `-Werror` in the CI?
> 
> I think we should, especially since vt's `vt-tv` pipeline was failing with `-Werror`. @pierrepebay @tlamonthezie Do you have any thoughts?

I think so, @tlamonthezie once this is merged could you add this parameter in your `build.sh` in #85, and enable it when calling it in CI?

**Review by nlslatt — APPROVED** [review]

**Pull request merged**